quinn-proto/src/connection/streams/mod.rs

```diff
@@ -79,6 +80,17 @@ impl<'a> Streams<'a> {
     pub fn send_streams(&self) -> usize {
         self.state.send_streams
     }
+
+    /// The number of remotely initiated open streams of a certain directionality.
+    ///
+    /// Includes remotely initiated streams, which have not been accepted via [`accept`](Self::accept).
+    /// These streams count against the respective concurrency limit reported by
+    /// [`Connection::max_concurrent_streams`](super::Connection::max_concurrent_streams).
+    pub fn remote_open_streams(&self, dir: Dir) -> u64 {
```

Comment:
What's the use case for this? It seems difficult to use, since it might become out of date immediately.

Reply:
The motivation is quite similar to the other one. It is hard to keep track of what is actually considered open and counts against the limit without knowing a bunch of implementation details. They complement each other nicely if you want to know the full picture of the current concurrency situation.

`max_concurrent_streams()` tells what the concurrency limit is right now.
`remote_open_streams()` tells you how much of it is in use.
The difference tells you how many free slots are left or if the remote hit the limit.

I personally don't need this one right now, so we can leave it for later. But knowing if the remote hit the limit is useful to make resource allocation decisions (especially if downstream resources are costly and need to be allocated/requested before advertising their availability upstream by increasing the concurrency limit).

Reply:
I usually prefer to limit API expansion to stuff that's immediately applicable, lest we end up committed to supporting a bunch of stuff that nobody ends up wanting. On the other hand, this does seem like a straightforward statistic, and it's hard to imagine it ever becoming a burden to expose. In the end I think I'm happy to have it, especially if @djc's not raising any eyebrows.